### PR TITLE
fix: get_writer errors if filename is none

### PR DESCRIPTION
**Pull request by jandom**
Fixes a small bug

Changes made in this Pull Request:
 - catch a case when filename is truthy and format is uninitialized 

PR Checklist
------------
 - [x] Tests? Added a simple unit test, maybe too much mocking, please review
 - [x] Docs? Not applicable
 - [x] CHANGELOG updated? Yes!
 - [x] Issue raised/referenced? Nope

### CODE REVIEW

File: package/CHANGELOG [hunk 1]
```diff
@@ -19,6 +19,7 @@ The rules for this file:
  * 2.5.0
 
 Fixes
+  * Fix get_writer format uninitialized (PR #4043)
```
Comment (orbeckst) [inline package/CHANGELOG]: Also add your GH handle to the list of contributors.

**Review by orbeckst — COMMENTED** [review]
What's the bug, i.e., do you have example code that triggers it?

**Comment by jandom** [comment]
It's a mini-bug that doesn't actually happen in the way the function is called. 

However, if you call the function with just the format argument, it'll error

```
selections.get_writer(None, "blah") 
```

This will error

**Comment by richardjgowers** [comment]
7cpwc6

File: package/CHANGELOG (hunk 1 above)
Comment (orbeckst) [reply package/CHANGELOG]: still todo

**Review by orbeckst — CHANGES_REQUESTED** [review]
lgtm, just add your GH handle to CHANGELOG

File: package/CHANGELOG (hunk 1 above)
Comment (IAlibay) [reply package/CHANGELOG]: I'm not going to make this a blocking request - but for the few folks that read the changelog to work out what changed in the package, could I ask for a little bit more information? Something like "Fix uninitialized `format` variable issue when calling `selections.get_writer directly`" would probably go a long way.

Comment (orbeckst) [reply package/CHANGELOG]: Good point, especially when we don't have an issue to point to.

Comment (jandom) [reply package/CHANGELOG]: Gotcha, on it

**Review by orbeckst — DISMISSED** [review]
Just a more explicit CHANGELOG entry, please, see https://github.com/MDAnalysis/mdanalysis/pull/4043/files#r1122512703

**Comment by orbeckst** [comment]
Not sure why CI is not running.

**Review by richardjgowers — APPROVED** [review]

**Comment by jandom** [comment]
Thanks for the merge, it's a small contribution but I hope to send more in

**Pull request merged**